packages/hardware-scan/default.nix

```diff
@@ -0,0 +1,50 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# This is a temporary solution for hardware detection.
+#
+{
+  stdenv,
+  util-linux,
+  pciutils,
+  usbutils,
+  dmidecode,
+  alejandra,
+  makeWrapper,
+  lib,
+}: let
+  inherit (lib) makeBinPath;
+  pname = "hardware-scan";
```

Comment:
would you not just use `writeshellApplication` and read in the file?

`text = builtins.readFile ./hardware-scan.sh;`

Reply:
changed, including file extension so shellcheck picks it up